nucliadb/nucliadb/ingest/maindb/driver.py

```diff
@@ -85,3 +86,16 @@ async def keys(
         self, match: str, count: int = DEFAULT_SCAN_LIMIT, include_start: bool = True
     ) -> AsyncGenerator[str, None]:
         raise NotImplementedError()
+
+    @asynccontextmanager
+    async def managed_transaction(self) -> AsyncGenerator[Transaction, None]:
+        """
+        Use to make sure transaction is always aborted
+        """
+        txn: Optional[Transaction] = None
+        try:
+            txn = await self.begin()
+            yield txn
+        finally:
+            if txn is not None:
+                await txn.abort()
```

Comment:
If the trasaction has been commited, will it fail when trying to abort on TiKV?

Reply:
good catch! It would not have failed, but we would have an unnecessary `await txn.abort()`